### PR TITLE
fix: reconnect when specified error occur with sftp connection

### PR DESCRIPTION
**Pull request by fengxsong**
Signed-off-by: fengxsong <fengxsong@outlook.com>

Support reconnect sftp client to prevent from disrupting the entire run process. default max retry times is 5.

### CODE REVIEW

File: pkg/ssh/scp.go [hunk 1]
```diff
@@ -110,17 +125,19 @@ func (s *SSH) Copy(host, localPath, remotePath string) error {
...
 
 	return s.doCopy(sftpClient, host, localPath, remotePath, bar)
 }
 
+func isErrorWorthRetry(err error) bool {
+	return strings.Contains(err.Error(), "connection reset by peer") ||
+		strings.Contains(err.Error(), io.EOF.Error())
+}
+
 func (s *SSH) doCopy(client *sftp.Client, host, src, dest string, epu *progressbar.ProgressBar) error {
```
Comment (fanux) [inline pkg/ssh/scp.go:141]: doCopy->copy

File: pkg/ssh/scp.go [hunk 2]
```diff
@@ -44,19 +43,21 @@ func (s *SSH) RemoteSha256Sum(host, remoteFilePath string) string {
 	return remoteHash
 }
 
-// CmdToString is in host exec cmd and replace to spilt str
-func (s *SSH) CmdToString(host, cmd, spilt string) (string, error) {
-	data, err := s.Cmd(host, cmd)
-	str := string(data)
+func getOnelineResult(output string, sep string) string {
```
[the remark this answers is not in the document]
Comment (fengxsong) [reply pkg/ssh/scp.go:46]: If we are trying to replace '\r\n' and '\n' with some sep or others, we're trying to get one line result, right?

**Review by cuisongliu — APPROVED** [review]

File: pkg/ssh/scp.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (fanux) [reply pkg/ssh/scp.go:46]: Maybe the CmdToString function don't care about '\n' is better. The result returned by the command is handed over to the caller for processing

[the remark this answers is not in the document]
Comment (fengxsong) [reply pkg/ssh/scp.go:46]: > Maybe the CmdToString function don't care about '\n' is better. The result returned by the command is handed over to the caller for processing

LGTM, I will split those functions later.

**Pull request merged**